Content.Client/GameTicking/Managers/ClientGameTicker.cs

```diff
@@ -153,8 +162,35 @@ private void RoundEnd(RoundEndMessageEvent message)
             if (_window?.RoundId == message.RoundId)
                 return;
 
-            //This is not ideal at all, but I don't see an immediately better fit anywhere else.
-            _window = new RoundEndSummaryWindow(message.GamemodeTitle, message.RoundEndText, message.RoundDuration, message.RoundId, message.AllPlayersEndInfo, _entityManager);
+            //Cyberfinn changes:
+            //back up the round info:
+            _roundEndContainer._message = message;
+            //display the window
+            DisplayRoundEndSummary(_roundEndContainer._message);
+
+            //enable the button, so that users can review the summary after closing the window
+            EnableRoundEndSummaryButton();
+        }
+        //Cyberfinn changes: moved this out to its own method
+        public static void DisplayRoundEndSummary(RoundEndMessageEvent? message)
+        {
+            if (message != null)
+                _window = new RoundEndSummaryWindow(message.GamemodeTitle, message.RoundEndText, message.RoundDuration, message.RoundId, message.AllPlayersEndInfo, _entityManager);
+        }
+
+        private void EnableRoundEndSummaryButton()
+        {
+            //enable the button, so we can still click it once we've closed round-end window - until next round starts, where we'll disable it again
+            if (EscapeUIController._escapeWindow != null)
+                if (EscapeUIController._escapeWindow.buttonShowRoundEnd != null)
+                    EscapeUIController._escapeWindow.buttonShowRoundEnd.Visible = true;
```

Comment:
replace these 3 lines with EscapeUIController?._escapeWindow?.buttonShowRoundEnd.Visible = true;

Reply:
I thought it safer to do the check that the window object exists, then check that the actual button exists, didnt want a null object reference exception, but will remove as suggested. Thanks again!

Reply:
so I tried adding your change, but it's complaining that EscapeUiC is a type, making it invalid in the context when trying to make it nullable.
I did previously make the EscapeUiC member "_escapeWindow" nullable though, so it's kinda halfway to what you wanted lol. I might leave as is though, since it works fine - although is just a bit verbose with the few extra logic checks.